### PR TITLE
Bump astroid to 3.3.7

### LINKED ISSUE #10112: Unable to import 'collections.abc'  using python 3.13.1
### Bug description

```python
import collections.abc
```

### Configuration

```ini

```

### Command used

```shell
We use a complex github actions but notice the change when python updated to 3.13.1
using pylint-3.3.2
```

### Pylint output

```python
************* Module spinn_utilities.checker
spinn_utilities/checker.py:1:0: C0304: Final newline missing (missing-final-newline)
spinn_utilities/checker.py:1:0: E0401: Unable to import 'collections.abc' (import-error)
spinn_utilities/checker.py:1:0: W0611: Unused import collections.abc (unused-import)
```

### Expected behavior

Not seeing the Unable to import 'collections.abc' (import-error)

Other messages are due to the tiny example

likely due to
https://github.com/python/cpython/commit/d46d291bd390027811937bbf5af2e2600b8398a9

### Pylint version

```shell
Python 3.13.1

Did not happen with 
Python 3.13.0
```

### OS / Environment

tested with github actions using 
Ubuntu
  22.0[4](https://github.com/SpiNNakerManchester/SpiNNUtils/actions/runs/12199055579/job/34032094669#step:1:4).5
  LTS

### Additional dependencies

```python

```

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description
Closes #10112

### CODE REVIEW

File: tests/functional/n/no/no_member_imports.py [hunk 1]
```diff
@@ -33,9 +33,9 @@ def test_ignored_modules_root_one_applies_as_well() -> None:
 
 
 def test_ignored_modules_patterns() -> None:
-    import collections
+    import importlib
 
-    collections.abc.THIS_does_not_EXIST
+    importlib.metadata.THIS_does_not_EXIST
```
Comment (jacobtylerwalls) [inline tests/functional/n/no/no_member_imports.py:38]: collections.abc is now an attribute. i don't love it, but python forced our hand. collections is not essential to this test. someone might file an issue eventually.

**Comment by jacobtylerwalls** [comment]
@Pierre-Sassoulas I don't get these failures locally. Do you?

**Comment by Pierre-Sassoulas** [comment]
I'm looking into it, right now tests aren't even green on 3.12, I'm going to cleanup my virtualenv

**Comment by Pierre-Sassoulas** [comment]
I have a single fail on 3.13:
```
_____________________________________________________________________ test_functional[unbalanced_tuple_unpacking] ______________________________________________________________________

self = <pylint.testutils.lint_module_test.LintModuleTest object at 0x75ad5c1a7e70>

    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong message(s) raised for "unbalanced_tuple_unpacking.py":
E       
E       Unexpected in testdata:
E        126: no-value-for-parameter
```

**Comment by jacobtylerwalls** [comment]
Deleted caches, retrying

**Comment by jacobtylerwalls** [comment]
Just checking, python 3.13.1?

**Comment by Pierre-Sassoulas** [comment]
(Right it was 3.13.0 🤦 , it seems 3.13.1 is not available with uv)

**Comment by Pierre-Sassoulas** [comment]
How do you install python 3.13.1 ? It seems my only option is docker or building it from source on linux :(

**Comment by jacobtylerwalls** [comment]
Python.org installer. Swear by them.

**Comment by Pierre-Sassoulas** [comment]
Hmm, ok thank you. I went to that too, but there's no compiled option for Linux it seems ? 



I'm a little afraid of "compiling it wrong". Might use the docker then.

**Comment by jacobtylerwalls** [comment]
Got it, forgot you develop on linux!

**Pull request closed without merging**